app/src/main/java/net/gsantner/markor/activity/DocumentEditAndViewFragment.java

```diff
@@ -132,9 +137,23 @@ public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         final Activity activity = getActivity();
 
         _hlEditor = view.findViewById(R.id.document__fragment__edit__highlighting_editor);
+        _editorHolder = view.findViewById(R.id.document__fragment__edit__editor_holder);
         _textActionsBar = view.findViewById(R.id.document__fragment__edit__text_actions_bar);
         _webView = view.findViewById(R.id.document__fragment_view_webview);
         _primaryScrollView = view.findViewById(R.id.document__fragment__edit__content_editor__scrolling_parent);
+        _primaryScrollView.getViewTreeObserver().addOnScrollChangedListener(() ->
+        {
+            if (_lineNumbersView.isLineNumbersEnabled()) {
+                final long time = System.currentTimeMillis();
+                if (time - _lineNumbersRefreshTime > 125) {
```

Comment:
Why a time based vs a scroll based refresh like we do now?

Reply:
Add a time based way just reduces the refresh rate of the line numbers when scrolling

Reply:
But we already have a system where the line numbering only redraws after the user has scrolled by some delta (1 page iirc). IMO that is superior

Reply:
That system is called passively when the view is changed, it works well when the view is EditText, but now this view is TextView. So this code is designed to actively trigger that system to check and determine whether to draw new line numbers.